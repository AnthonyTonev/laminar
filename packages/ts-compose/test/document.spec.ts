import {
  document,
  DocumentContext,
  printDocument,
  Type,
  Node,
  withIdentifier,
  withImports,
} from '../src';

describe('Document', () => {
  it('Test print document with context', () => {
    const initial: DocumentContext = { imports: {}, identifiers: {} };

    const context = withImports(
      withIdentifier(
        withIdentifier(initial, Type.Alias({ name: 'test', type: Type.Str })),
        Node.Const({ name: 'a', value: { test1: 'other', k: 12 }, multiline: true }),
      ),
      'utils',
      ['inspect', 'tmp'],
    );

    const doc = document(
      context,
      Type.Interface({
        name: 'Orange',
        props: [
          Type.Prop({ name: 'value', type: Type.Num }),
          Type.Prop({ name: 'other', type: Type.Ref('test') }),
        ],
      }),
    );

    const source = printDocument(doc);
    expect(source).toMatchSnapshot();
  });
});
