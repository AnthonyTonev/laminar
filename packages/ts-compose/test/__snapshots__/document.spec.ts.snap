// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Test print document with context 1`] = `
"import { inspect, tmp } from \\"utils\\";

interface Orange {
    value: number;
    other: test;
}

type test = string;

const a = {
    test1: \\"other\\",
    k: 12
};"
`;
